webdriver/tests/bidi/__init__.py

```diff
@@ -79,3 +79,43 @@ async def create_console_api_message(bidi_session, context, text):
         target=ContextTarget(context["context"]),
     )
     return text
+
+
+async def get_device_pixel_ratio(bidi_session, context):
+    """Get the DPR of the context.
+
+    :param bidi_session: BiDiSession
+    :param context: Browsing context ID
+    :returns: (int) devicePixelRatio.
+    """
+    result = await bidi_session.script.call_function(
+        function_declaration="""() => {
+        return Math.floor(window.devicePixelRatio);
+    }""",
+        target=ContextTarget(context["context"]),
+        await_promise=False)
+    return result["value"]
+
+
+async def get_viewport_dimensions(bidi_session, context):
+    expression = """
+        ({
+          height: window.innerHeight || document.documentElement.clientHeight,
+          width: window.innerWidth || document.documentElement.clientWidth,
+        });
+    """
+    result = await bidi_session.script.evaluate(
+        expression=expression,
+        target=ContextTarget(context["context"]),
+        await_promise=False,
+    )
+
+    return remote_mapping_to_dict(result["value"])
+
+
+def remote_mapping_to_dict(js_object):
+    obj = {}
```

Comment:
```suggestion
def remote_mapping_to_dict(js_object: Dict) -> Dict:
```

(please double-check)

Reply:
Follow-up will live here: https://github.com/web-platform-tests/wpt/pull/40643/files